onnxruntime/core/providers/cpu/tensor/space_depth_ops.cc

```diff
@@ -16,103 +16,132 @@ ONNX_CPU_OPERATOR_VERSIONED_KERNEL(
     SpaceToDepth,
     1,
     12,
-    KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
-    SpaceToDepth<float>);
+    KernelDefBuilder()
+        .TypeConstraint("T", {DataTypeImpl::GetTensorType<float>(),
+                              DataTypeImpl::GetTensorType<double>()}),
```

Comment:
can support double - I think it has more advantages than dis-advantages. The binary size increase should be very marginal as the templated code section is very small.